x/evm/handler.go

```diff
@@ -22,6 +22,9 @@ func NewHandler(server types.MsgServer) sdk.Handler {
 		case *types.MsgEthereumTx:
 			// execute state transition
 			res, err := server.EthereumTx(sdk.WrapSDKContext(ctx), msg)
+			if err == nil {
+				SetGasInContext(ctx, res.GasUsed)
```

Comment:
Why not set this regardless of the error?

Reply:
if there is an error, the value from res.GasUsed can be wrong or not even set. In that case not sure how to handle it.

Reply:
the gas used is returned to a user regardless if the tx was executed or not. I think we should set it. 

It's unfortunate that the context is not a pointer value on the SDK, would make things much easier

Reply:
the problem is if there is an error `res` will be null, `res.GasUsed` would trigger a nil pointer exception

Reply:
I can either return the real gas used (which is what I am doing now) or I can set the gas used to 0